Source/Moya+Defaults.swift

```diff
@@ -3,8 +3,12 @@ import Alamofire
 /// These functions are default mappings to `MoyaProvider`'s properties: endpoints, requests, manager, etc.
 public extension MoyaProvider {
     public final class func DefaultEndpointMapping(target: Target) -> Endpoint<Target> {
-        let url = target.baseURL.URLByAppendingPathComponent(target.path).absoluteString
-        return Endpoint(URL: url, sampleResponseClosure: {.NetworkResponse(200, target.sampleData)}, method: target.method, parameters: target.parameters)
+        if let url = target.baseURL.URLByAppendingPathComponent(target.path) {
```

Comment:
In that case, I'd do the following:

```swift
if let
    url = target.baseURL.URLByAppendingPathcomponent(target.path),
    stringURL = url.absolutestring {
        return Endpoint(URL: url, sampleResponseClosure: { .NetworkResponse(200, target.sampleData) }, method: target.method, parameters: target.parameters)
}
fatalError("\(target) url is not valid")
```

Notes:
- `url` shouldn't need force unwrapping, as it has been unwrapped previously, right?
- we might want to go with an inverted code path here, using `guard` to fail early.